[DNM] clusterversion: set devBranch to false, version to beta.1

celiala opened the pull request:
Release note: None
Epic: None

celiala committed:
clusterversion: set devBranch to false, version to beta.1

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/122044)
<!-- Reviewable:end -->

celiala committed:
demo: disable multitenancy by default

Backport 1/1 commits from #112305.

We need to do this as part of setting developmentBranch to false for
release branches.

In every release since 22.2 we've enabled multitenancy by default on the master
branch for cockroach demo. This has resulted in much more robust user testing,
and has uncovered several bugs. We're still however, not at the point where our
multi-tenant UX completely matches our single-tenant UX. As a result, before we
ship 23.2 we're going to disable multitenancy by default for demo.

Epic: none

Release note: None
Release justification: Reverting functional regression.
